src/user.ts

```diff
@@ -51,27 +51,32 @@ export type UserStorage = {
     accountName: Name;
     username: TonomyUsername;
     salt: Checksum256;
+    // TODO update to have all data from blockchain
 };
 
 const idContract = IDContract.Instance;
 
 export class User {
     keyManager: KeyManager;
     storage: UserStorage & PersistentStorageClean;
-    app: App;
+    apps: UserApps;
 
     constructor(_keyManager: KeyManager, storageFactory: StorageFactory) {
         this.keyManager = _keyManager;
         this.storage = createStorage<UserStorage>('tonomy.user.', storageFactory);
 
-        this.app = new App(this, _keyManager, storageFactory);
+        this.apps = new UserApps(this, _keyManager, storageFactory);
```

Comment:
this should be injected instead of constructed  inside the class.
to make sure we are using dependency inversion
https://stackify.com/dependency-inversion-principle/

Reply:
https://github.com/Tonomy-Foundation/Tonomy-ID-SDK/blob/feature/222-app-login-screen-consent-logic/src/userApps.ts#L52
also L68

UserApp is always a sub-object of User object
therefore I don't see any value in doing this